packages/react/src/components/AccountSettings/DeleteUser/DeleteUser.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+import { Logger } from 'aws-amplify';
+import { deleteUser, translate } from '@aws-amplify/ui';
+
+import { useAuth } from '../../../internal';
+import { Flex } from '../../../primitives';
+import {
+  DefaultWarning,
+  DefaultError,
+  DefaultSubmitButton,
+} from './defaultComponents';
+import { DeleteUserProps, DeleteUserState } from './types';
+
+const logger = new Logger('DeleteUser');
+
+function DeleteUser({
+  onSuccess,
+  onError,
+  handleDelete,
+}: DeleteUserProps): JSX.Element | null {
+  const [state, setState] = React.useState<DeleteUserState>('IDLE');
+  const [errorMessage, setErrorMessage] = React.useState<string>(null);
+
+  // translations
+  const deleteAccountText = translate('Delete Account');
+
+  const { user, isLoading } = useAuth();
+
+  const startConfirmation = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    setState('IS_CONFIRMING');
+  };
+
+  const runDeleteUser = React.useCallback(async () => {
+    setState('IN_PROGRESS');
+    try {
+      if (handleDelete) {
+        await handleDelete(user);
+      } else {
+        await deleteUser();
```

Comment:
Just checking that we don't need to pass `user` to `deleteUser`?

Reply:
No, because `Auth.deleteUser` actually does not require underlying `user` object.

Reply:
On the other hand, if customer provided their own `handleDelete` function to override the default service, we will pass `user` object for their convenience so that they can do whatever clean up with the user.

Reply:
I do not think the code is clear here, i can add a comment (or let me know if you have any refactor suggestions)

Reply:
Comments added in https://github.com/aws-amplify/amplify-ui/pull/2927/commits/9df0d8c22af6f0570d3e554d7a49b872b482cf29, resolving.